### PR TITLE
Add warning if IntersectionObserver API is not available

### PR DESCRIPTION
**Pull request by vega-kuromana**
## Description
- Add warning when no IntersectionObserver API 
- Do nothing in `created`, `mounted` and `destroyed` when no IntersectionObserver API

### CODE REVIEW

File: __tests__/index.js [hunk 1]
```diff
@@ -217,3 +217,52 @@ test('It should not warn if Vue.config.silent is set to false', async () => {
 
   global.console.warn.mockReset()
 })
+
+test('It should warn if IntersectionObserver is not available', async () => {
```
Comment (marocchino) [inline __tests__/index.js]:
```js
describe('when no IntersectionObserver', () => {
  beforeEach(() => {
    global.IntersectionObserver = undefined
  })

  afterEach(() => {
    global.IntersectionObserver = IntersectionObserver
  })
})
```

File: __tests__/index.js [hunk 2]
```diff
@@ -217,3 +217,52 @@ test('It should not warn if Vue.config.silent is set to false', async () => {
...
+  global.IntersectionObserver = undefined
+  global.console.warn = jest.fn()
+  const vm = new Vue({
+    template: `<intersect><div></div><div></div></intersect>`,
+    components: {Intersect}
+  }).$mount()
+
+  await vm.$nextTick()
+
+  expect(global.console.warn).toHaveBeenCalled()
```
Comment (marocchino) [inline __tests__/index.js]: not needed.

File: __tests__/index.js [hunk 3]
```diff
@@ -217,3 +217,52 @@ test('It should not warn if Vue.config.silent is set to false', async () => {
...
+  vm.observer = {
+    observe: jest.fn()
+  }
+
+  expect(vm.$nextTick).not.toHaveBeenCalled()
+  expect(vm.observer.observe).not.toHaveBeenCalled()
+  global.IntersectionObserver = IntersectionObserver
+})
+
+test('It should return in destroyed function if IntersectionObserver is not available', () => {
```
Comment (marocchino) [inline __tests__/index.js]: return what?
> It should not disconnect when the component is destroyed

File: dist/index.js [hunk 4]
```diff
@@ -52,6 +58,7 @@ export default {
   mounted: function mounted() {
     var _this2 = this;
 
+    if (!this.isIntersectionObserver) return;
```
Comment (marocchino) [inline dist/index.js]: this -> _this2

File: __tests__/index.js [hunk 5]
```diff
@@ -217,3 +217,52 @@ test('It should not warn if Vue.config.silent is set to false', async () => {
...
+  }).$mount()
+
+  await vm.$nextTick()
+
+  expect(global.console.warn).toHaveBeenCalled()
+  expect(global.console.warn).toHaveBeenCalledWith('[vue-intersect] IntersectionObserver API is not available in your browser. Please install this polyfill: https://github.com/WICG/IntersectionObserver/tree/gh-pages/polyfill')
+  global.IntersectionObserver = IntersectionObserver
+})
+
+test('It should return in mounted function if IntersectionObserver is not available', () => {
```
Comment (marocchino) [inline __tests__/index.js]: It should not observe when the component is mounted

File: __tests__/index.js [hunk 6]
```diff
@@ -217,3 +217,52 @@ test('It should not warn if Vue.config.silent is set to false', async () => {
...
+})
+
+test('It should return in destroyed function if IntersectionObserver is not available', () => {
+  global.IntersectionObserver = undefined
+  const vm = new Vue({
+    template: `<intersect><div></div><div></div></intersect>`,
+    components: {Intersect}
+  }).$mount()
+
+  vm.observer = {
```
Comment (marocchino) [inline __tests__/index.js]: wrong mock: No observer object is being in real case.

File: src/index.js [hunk 7]
```diff
@@ -27,6 +27,12 @@ export default {
     }
   },
   created () {
+    this.isIntersectionObserver = typeof IntersectionObserver !== 'undefined'
```
Comment (marocchino) [inline src/index.js]: not test this.

**Comment by vega-kuromana** [comment]
Fixed. @marocchino

**Review by marocchino — APPROVED** [review]

**Pull request merged**